### PR TITLE
Add drafting

### PR DESCRIPTION
**Pull request by afieldi**
I think drafting now works:

Here is an example finished draft: https://risen-stats-git-draft-areks-projects-ad5b7dce.vercel.app/drafting/vi0mmKjK57/5Q9I4Gl9OG

While draft is in progress it will be kept in-mem. Then when it's finished it will be removed from mem and saved to the DB. I hope it works :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (afieldi) [inline package.json:1]: Needed for heroku I think

File: BackEnd/src/business/drafting.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      auth: MakeId(10),
+      displayName: redName,
+    },
+    stage: 0,
+    roomId: MakeId(10),
+    timerMax: DEFAULT_DRAFT_TIME,
+    timerRemaining: DEFAULT_DRAFT_TIME,
+    roomActive: false,
+  };
+  gameCache.set(draftState.roomId, draftState, 60 * 60 * 12); // Keep drafts in memory for 12 hours
```
Comment (Vexrax) [inline BackEnd/src/business/drafting.ts:55]: What happens if a draft is in proccess when we deploy or something happens to the server, is it lost to the void?

File: Common/utils.ts [hunk 3]
```diff
@@ -320,4 +321,26 @@ export function getSeasonWithLeaguePage(seasons: SeasonModel[], seasonId?: strin
...
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  let counter = 0;
+  while (counter < length) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    counter += 1;
+  }
+  return result;
+}
```
Comment (Vexrax) [inline Common/utils.ts:336]: Im pretty sure theres a better way to do this, but ig this works for now. IIRC normally people filter out bad words etc.

File: Common/utils.ts [hunk 4]
```diff
@@ -320,4 +321,26 @@ export function getSeasonWithLeaguePage(seasons: SeasonModel[], seasonId?: strin
...
+
+export function getTeamFromDraftState(game: DraftState, auth: string): 'blueTeam' | 'redTeam' {
+  if (game.blueTeam.auth === auth) {
+    return 'blueTeam';
```
Comment (Vexrax) [inline Common/utils.ts:340]: Seems like ur using the `blueTeam` and `redTeam` consts more than once, does this need to be defined in a var/type so we only need to make changes in one spot?

File: front-end/src/components/drafting/champion-picker.tsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  const unavailableChamps: Set<string> = useMemo(() => {
+    if (!draftState) return new Set();
+    const team = getTeamFromDraftState(draftState, auth ?? '');
+    const champs = [
+      ...draftState.blueTeam.picks,
+      ...draftState.blueTeam.bans,
+      ...draftState.redTeam.picks,
+      ...draftState.redTeam.bans,
+    ];
+    if (team === 'redTeam') {
```
Comment (Vexrax) [inline front-end/src/components/drafting/champion-picker.tsx:69]: Same comment as above, does this need a type?

File: front-end/src/components/drafting/champion-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+      champs.push(...draftState.redTeam.disabledPicks);
+    }
+    else if (team === 'blueTeam') {
```
Comment (Vexrax) [inline front-end/src/components/drafting/champion-picker.tsx:72]: Same comment as above, does this need a type?

File: front-end/src/draftSocket.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+import { DraftingSocketClientToServer, DraftingSocketServerToClient } from '../../Common/Interface/Internal/drafting';
+
+// const socket: Socket<
+//   DraftingSocketServerToClient,
+//   DraftingSocketClientToServer
+// > = io(process.env.REACT_APP_BACKEND_URL, {
+//   path: DRAFT_SOCKET_PATH,
+// });
+//
+// export default socket;
```
Comment (Vexrax) [inline front-end/src/draftSocket.ts:12]: Do we need this file?

**Comment by Vexrax** [comment]
Lets do a demo so i can understand what the draft proccess looks like?

File: BackEnd/data/champions_map.json
[the remark this answers is not in the document]
Comment (afieldi) [reply BackEnd/data/champions_map.json:1]: Yep. The update script

File: BackEnd/src/business/drafting.ts (hunk 2 above)
Comment (afieldi) [reply BackEnd/src/business/drafting.ts:55]: Yeah it's lost in the void.

File: Common/utils.ts (hunk 4 above)
Comment (afieldi) [reply Common/utils.ts:340]: Tbh probably should be lol

File: front-end/src/draftSocket.ts (hunk 7 above)
Comment (afieldi) [reply front-end/src/draftSocket.ts:12]: No

**Review by Vexrax — APPROVED** [review]
LGTM

**Pull request merged**